src/rust/lib/graph-editor/src/lib.rs

```diff
@@ -356,16 +356,17 @@ pub struct FrpInputs {
     pub remove_node                  : frp::Source<NodeId>,
     pub set_node_expression          : frp::Source<(NodeId,node::Expression)>,
     pub set_node_position            : frp::Source<(NodeId,Position)>,
-    pub set_visualization_data       : frp::Source<NodeId>,
     pub translate_selected_nodes     : frp::Source<Position>,
     pub cycle_visualization          : frp::Source<NodeId>,
     pub set_visualization            : frp::Source<(NodeId,Option<Visualization>)>,
     pub register_visualization_class : frp::Source<Option<Rc<visualization::Handle>>>,
-
+    pub set_visualization_data       : frp::Source<(NodeId,Option<visualization::Data>)>,
 
     hover_node_input           : frp::Source<Option<EdgeTarget>>,
     some_edge_targets_detached : frp::Source,
     all_edge_targets_attached  : frp::Source,
+    on_visualization_enabled   : frp::Source<NodeId>,
+    on_visualization_disabled  : frp::Source<NodeId>
```

Comment:
Instead of crerating these two FRP endpoints in inputs, just create them locally. They do not need to be remembered in theinputs struct. Just create them in the place where you use them instead.